flink-table-store-kafka/src/test/java/org/apache/flink/table/store/kafka/KafkaLogITCase.java

```diff
@@ -0,0 +1,189 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.kafka;
+
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.factories.DynamicTableFactory.Context;
+import org.apache.flink.table.store.log.LogOptions.LogChangelogMode;
+import org.apache.flink.table.store.log.LogOptions.LogConsistency;
+import org.apache.flink.types.RowKind;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.UUID;
+
+import static org.apache.flink.streaming.api.environment.ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH;
+import static org.apache.flink.table.store.kafka.KafkaLogTestUtils.SINK_CONTEXT;
+import static org.apache.flink.table.store.kafka.KafkaLogTestUtils.SOURCE_CONTEXT;
+import static org.apache.flink.table.store.kafka.KafkaLogTestUtils.discoverKafkaLogFactory;
+import static org.apache.flink.table.store.kafka.KafkaLogTestUtils.testContext;
+import static org.apache.flink.table.store.kafka.KafkaLogTestUtils.testRecord;
+
+/** ITCase for {@link KafkaLogStoreFactory}. */
+public class KafkaLogITCase extends KafkaTableTestBase {
+
+    private final KafkaLogStoreFactory factory = discoverKafkaLogFactory();
+
+    @Test
+    public void testDropEmpty() {
+        factory.onDropTable(testContext(getBootstrapServers(), LogChangelogMode.AUTO, true));
+    }
+
+    @Test
+    public void testUpsertTransactionKeyed() throws Exception {
+        innerTest(
+                "UpsertTransactionKeyed",
+                LogChangelogMode.UPSERT,
+                LogConsistency.TRANSACTIONAL,
+                true);
+    }
+
+    @Test
+    public void testAllTransactionKeyed() throws Exception {
+        innerTest("AllTransactionKeyed", LogChangelogMode.ALL, LogConsistency.TRANSACTIONAL, true);
+    }
+
+    @Test
+    public void testUpsertEventualKeyed() throws Exception {
+        innerTest("UpsertEventualKeyed", LogChangelogMode.UPSERT, LogConsistency.EVENTUAL, true);
+    }
+
+    @Test
+    public void testAllEventualKeyed() throws Exception {
+        innerTest("AllEventualKeyed", LogChangelogMode.ALL, LogConsistency.EVENTUAL, true);
+    }
+
+    @Test
+    public void testAllTransactionNonKeyed() throws Exception {
+        innerTest(
+                "AllTransactionNonKeyed",
+                LogChangelogMode.ALL,
+                LogConsistency.TRANSACTIONAL,
+                false);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
Nit: we'd better use `ExpectedException` to check the expected message as well.

Reply:
It is recommended to use `Assert.assertThrows`.